cypress/support/commands.ts

```diff
@@ -58,3 +61,73 @@ Cypress.on('uncaught:exception', (err: Error): boolean | Error => {
 
   return err;
 });
+
+const toJsonDate = (date: Date): string => date.toJSON().split('T')[0];
+
+Cypress.Commands.add(
+  'createDatePeriod',
+  ({
+    name,
+    startDate,
+    endDate,
+    resourceId,
+  }: {
+    name: string;
+    startDate: Date;
+    endDate: Date;
+    resourceId: string;
+  }): Chainable => {
+    return cy
+      .visit('/')
+      .window()
+      .then((win: Window) => {
+        const apiUrl = win.ENV?.API_URL;
+
+        if (!apiUrl) {
+          return cy.task('log', 'API_URL is not set!!!');
```

Comment:
I wonder if we could reduce the amount of nested promise chains by throwing an error here, and then enclosing this then branch?

Reply:
It throws an error now